api/v3/Case.php

```diff
@@ -101,31 +131,46 @@ function civicrm_api3_case_create($params) {
     CRM_Case_BAO_CaseContact::create($contactParams);
   }
 
+  if (!isset($params['id'])) {
+    // Creating a new case
```

Comment:
We are, I don't really understand the xml stuff very well, except that cases are defined in xml files, and when creating a new one you have to read the xml file to get case params/setup the case.  I've added a comment to that effect - the code in _civicrm_api3_case_create_xmlProcessor is simply extracted code from the original api_v3_case_create function.